### ISSUE TITLE
Integration of the Deepmd Plugin the the latest verion of DeePMD-kit

### ISSUE THREAD

**Issue by dingye18** [JingHuangLab/openmm_deepmd_plugin#25]
As DeePMD-kit v3 support pytorch. Add support for the pytorch backend is needed for OpenMM Deepmd Plugin.

- [ ] DeePMD-kit v3 support
- [ ] Unit test coverage
- [ ] Conda recipe for this plugin.

**Comment by KingOfChuXuan** [comment]
Hi team,

Just following up on this issue. I was wondering if there are any updates or if this is currently being looked into?

Thank you so much for your time and for maintaining this helpful plugin!

**Comment by dingye18** [comment]
Apologies for the delayed update of this plugin. It currently supports DeePMD-kit v3 and later versions. If you encounter any issues while using it with DeePMD-kit v3, please let me know. Thank you.

**Comment by KingOfChuXuan** [comment]
Hi, I’ve re-installed the plugin in a fresh environment, but I’m getting this error when loading a `.pth` model:
`Exception: DeePMD-kit C API Error: DeePMD-kit Error: PyTorch backend is not built`

It seems the plugin failed to detect or enable the PyTorch backend during the compilation process. Could you please tell me if I need to pass any specific flags (e.g., `-DUSE_PYTORCH=ON` or similar) to `cmake`, or if there are any specific dependencies I need to have in my environment for the plugin to recognize the PyTorch backend of DeePMD-kit v3?

**Comment by dingye18** [comment]
It seems like an important issue. Could you provide an example .pth model so I can replicate the error and address it?

**Comment by KingOfChuXuan** [comment]
Thank you for looking into this! I have uploaded the test script to the repository:[run.py](https://github.com/user-attachments/files/28353310/run.py)
Regarding the model files (`.pth`), I encountered GitHub upload limitations due to the file size and security restrictions. However, to make it easy for you to replicate the issue, I have documented exactly how I generated these models from the source:
1. Original source: The `graph.pb` file is from the [Quantum-phase-diagram-of-water repository](https://github.com/sigbjobo/Quantum-phase-diagram-of-water/tree/main/Model/GraphFiles).
2. Conversion: I converted it using:
```dp convert-backend graph.pb graph.pth```
3. Compression: I then generated the compressed version using:
```dp --pt compress -i graph.pth -o graph-compress.pth```

By the way, my current deepmd-kit version is `3.1.3`.
You should be able to generate these models quickly using these commands in your environment. As I can't upload the `PDB` file, my test system consists of a 432-water-molecule box. 
Please let me know if you need any additional information to help debug the error.

Thank you again for your time and assistance!

**Comment by dingye18** [comment]
Certainly! I will review the operational details of this model using the DeepMD plugin.

**Comment by KingOfChuXuan** [comment]
Thank you for the update! I really appreciate you taking the time to look into this. I'll stay tuned for any findings or if you need any further information from my side during your testing. Looking forward to your progress!

**Comment by KingOfChuXuan** [comment]
Hi,

I hope you are doing well.

I just wanted to gently follow up on this issue to see if you've had a chance to look into if there is any progress on the PyTorch backend integration.

Since this block is currently critical for my ongoing project, please let me know if there are any specific compilation flags, environment variables, or temporary workarounds I could try on my end in the meantime.

Thank you so much for your time and continuous help!

**Comment by dingye18** [comment]
Apologies for the delayed response; I am currently working on it and hope to resolve it before this weekend.
